File: app/src/main/java/ru/kireev/mir/laregistrationmain/AddManuallyActivity.java
package ru.kireev.mir.laregistrationmain;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.Toast;

import ru.kireev.mir.laregistrationmain.data.MainViewModel;
import ru.kireev.mir.laregistrationmain.data.Volunteer;

public class AddManuallyActivity extends AppCompatActivity {

    private EditText editTextName;
    private EditText editTextSurname;
    private EditText editTextCallSign;
    private EditText editTextPhoneNumber;
    private CheckBox checkBoxHaveACar;
    private EditText editTextCarMark;
    private EditText editTextCarModel;
    private EditText editTextCarRegistrationNumber;
    private EditText editTextCarColor;
    private String name;
    private String surname;
    private String callSign;
    private String phoneNumber;
    private boolean haveACar;
    private String carMark;
    private String carModel;
    private String carRegistrationNumber;
    private String carColor;
    private MainViewModel mainViewModel;
    private int index;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_manually);

        editTextName = findViewById(R.id.editTextName);
        editTextSurname = findViewById(R.id.editTextSurname);
        editTextCallSign = findViewById(R.id.editTextCallSign);
        editTextPhoneNumber = findViewById(R.id.editTextPhoneNumber);
        checkBoxHaveACar = findViewById(R.id.checkBoxHaveACar);
        editTextCarMark = findViewById(R.id.editTextCarMark);
        editTextCarModel = findViewById(R.id.editTextCarModel);
        editTextCarRegistrationNumber = findViewById(R.id.editTextCarRegistrationNumber);
        editTextCarColor = findViewById(R.id.editTextCarColor);
        mainViewModel = new ViewModelProvider(this).get(MainViewModel.class);

        checkBoxHaveACar.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if (isChecked) {
                    editTextCarMark.setVisibility(View.VISIBLE);
                    editTextCarModel.setVisibility(View.VISIBLE);
                    editTextCarRegistrationNumber.setVisibility(View.VISIBLE);
                    editTextCarColor.setVisibility(View.VISIBLE);
                }
                else {
                    editTextCarMark.setVisibility(View.INVISIBLE);
                    editTextCarModel.setVisibility(View.INVISIBLE);
                    editTextCarRegistrationNumber.setVisibility(View.INVISIBLE);
                    editTextCarColor.setVisibility(View.INVISIBLE);
                }
            }
        });

        Intent intent = getIntent();
        index = intent.getIntExtra("size",0);
    }

    public void onClickSaveData(View view) {
        name = editTextName.getText().toString();
        surname = editTextSurname.getText().toString();
        callSign = editTextCallSign.getText().toString();
        phoneNumber = editTextPhoneNumber.getText().toString();
        haveACar = checkBoxHaveACar.isChecked();
        carMark = editTextCarMark.getText().toString();
        carModel = editTextCarModel.getText().toString();
        carRegistrationNumber = editTextCarRegistrationNumber.getText().toString();
        carColor = editTextCarColor.getText().toString();

        if (name.isEmpty() || surname.isEmpty() || phoneNumber.isEmpty()) {
            Toast.makeText(this, "Заполните поля Имя, Фамилия и Номер телефона!", Toast.LENGTH_SHORT).show();
        }
        else if (haveACar && (carMark.isEmpty() || carModel.isEmpty() || carRegistrationNumber.isEmpty() || carColor.isEmpty())) {
            Toast.makeText(this, "Заполните все данные об автомобиле!", Toast.LENGTH_SHORT).show();
        }
        else {
            Volunteer volunteer = new Volunteer(index, name, surname, callSign, phoneNumber, "false",
                        carMark, carModel, carRegistrationNumber, carColor, "false");

            mainViewModel.insertVolunteer(volunteer);
            onBackPressed();
        }

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        int id = item.getItemId();
        switch (id) {
            case R.id.menu_item_foxes:
                Intent intentFoxes = new Intent(this, FoxesMainActivity.class);
                startActivity(intentFoxes);
                break;
            case R.id.menu_item_volunteers:
                Intent intentVolunteers = new Intent(this, TabbedMainActivity.class);
                startActivity(intentVolunteers);
                break;
        }


        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/ru/kireev/mir/registrarlizaalert/AddManuallyActivity.kt
package ru.kireev.mir.registrarlizaalert

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_add_manually.*
import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking
import ru.kireev.mir.registrarlizaalert.data.Volunteer
import ru.kireev.mir.registrarlizaalert.data.VolunteersViewModel

class AddManuallyActivity : AppCompatActivity() {

    companion object {
        private const val EXTRA_SIZE = "size"
        private const val EXTRA_VOLUNTEER_ID = "volunteer_id"
    }

    private lateinit var viewModel: VolunteersViewModel
    private var index = 0
    private var volunteerId = 0
    private var isEdited = false
    private lateinit var volunteer: Volunteer

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_manually)
        val model by viewModels<VolunteersViewModel>()
        viewModel = model
        index = intent.getIntExtra(EXTRA_SIZE, 0)
        volunteerId = intent.getIntExtra(EXTRA_VOLUNTEER_ID, 0)
        if (volunteerId != 0) {
            runBlocking {
                isEdited = true
                val job = launch {
                    volunteer = viewModel.getVolunteerById(volunteerId)
                    editTextFullName.setText(volunteer.fullName)
                    editTextCallSign.setText(volunteer.callSign)
                    editTextForumNickname.setText(volunteer.nickName)
                    editTextRegion.setText(volunteer.region)
                    editTextPhoneNumber.setText(volunteer.phoneNumber)
                    editTextCar.setText(volunteer.car)
                }
                job.join()
            }
        }
    }

    fun onClickSaveData(view: View) {
        val fullName = editTextFullName.text.toString().trim()
        val callSign = editTextCallSign.text.toString().trim()
        val nickName = editTextForumNickname.text.toString().trim()
        val region = editTextRegion.text.toString().trim()
        val phoneNumber = editTextPhoneNumber.text.toString().trim()
        val car = editTextCar.text.toString().trim()

        if (fullName.isEmpty() || phoneNumber.isEmpty()) {
            Toast.makeText(this, getString(R.string.fill_in_fields_volunteer), Toast.LENGTH_SHORT).show()
        } else if (isEdited) {
            volunteer.fullName = fullName
            volunteer.callSign = callSign
            volunteer.nickName = nickName
            volunteer.region = region
            volunteer.phoneNumber = phoneNumber
            volunteer.car = car
            viewModel.insertVolunteer(volunteer)
            onBackPressed()
        } else {
            val status = getString(R.string.volunteer_status_active)
            val volunteer = Volunteer(0, index, fullName, callSign, nickName, region, phoneNumber, car, status = status)
            runBlocking {
                val job = launch {
                    if (viewModel.checkForVolunteerExist(fullName, phoneNumber)) {
                        Toast.makeText(this@AddManuallyActivity, getString(R.string.warning_qr_code_scan_volunteer_exist_message), Toast.LENGTH_SHORT).show()
                    } else {
                        viewModel.insertVolunteer(volunteer)
                        onBackPressed()
                    }
                }
                job.join()
            }
        }
    }
}